lib/generate.js

```diff
@@ -35,36 +37,64 @@ function dedentFunc(func) {
     });
 }
 
+function tabulateString(string, indent, filter) {
+    if(Array.isArray(string)) {
+      string = string.join('\n');
+    }
+    tabulated = string.toString().split('\n').map(function addIndent(line, i) {
+        if(filter) {
+            if(filter.call(line, i)) {
+                return indent + line;
+            } else {
+                return line;
+            }
+        } else {
+            return indent + line;
+        }
+    }).join('\n');
+    return tabulated;
+}
+
 function serializeSymbol(s) {
     return (s instanceof RegExp) ? s.toString()
                                  : JSON.stringify(s);
 }
 
+function id(a) {
+  return a;
+}
+
 function serializeRule(rule) {
     var ret = '{';
     ret += '"name": ' + JSON.stringify(rule.name);
     ret += ', "symbols": [' + rule.symbols.map(serializeSymbol).join(', ') + ']';
     if (rule.postprocess) {
-        var lines = dedentFunc(rule.postprocess);
-        ret += ', "postprocess": ' + lines.map(function indent(line, i) {
-            if (i > 0) {
-                line = '    ' + line;
-            }
-            return line;
-        }).join('\n');
+        if(rule.postprocess.special) {
+            rule.postprocess = this.specialPostprocessors[rule.postprocess.special];
```

Comment:
I think it would be better to have `specialPostprocessors` as an argument rather than something bound to `this`.